src/System.Globalization/tests/CultureInfo/CultureInfoCurrentCulture.cs

```diff
@@ -57,5 +58,38 @@ public void CurrentUICulture()
             }
             Assert.Equal(CultureInfo.CurrentUICulture, defaultUICulture);
         }
+
+        [Fact]
+        public static void DefaultThreadCurrentCulture()
+        {
+            CultureInfo originalCulture = CultureInfo.CurrentCulture;
+            CultureInfo originalUICulture = CultureInfo.CurrentUICulture;
+            CultureInfo originalDefaultThreadCurrentCulture = CultureInfo.DefaultThreadCurrentCulture;
+            CultureInfo originalDefaultThreadCurrentUICulture = CultureInfo.DefaultThreadCurrentUICulture;
+            CultureInfo jaCulture = new CultureInfo("ja-JP");
+
+            // Run the test only when we have a culture different than ja-JP
+            if (originalCulture.Equals(jaCulture) || originalUICulture.Equals(jaCulture))
+                return;
+
+            try
+            {
+                CultureInfo.DefaultThreadCurrentCulture = jaCulture;
+                CultureInfo.DefaultThreadCurrentUICulture = jaCulture;
+
+                Task.Run(() =>
+                {
+                    Assert.True(CultureInfo.CurrentCulture.Equals(jaCulture));
+                    Assert.True(CultureInfo.CurrentUICulture.Equals(jaCulture));
+                }).Wait();
```

Comment:
What's the goal of using Task.Run here?  Is it to ensure you're on another thread and that the culture flowed to that other thread?  If so, this usage of Wait() might thwart that.  When you Wait on a Task, if the Task hasn't started running yet, Wait might be able to take the task and execute it on the current thread, rather than blocking waiting for some other thread to eventually run it.  If you need to ensure that the Task definitely does not run on this thread, you have a few options; the easiest is probably to change this to be:
```C#
Task t = Task.Run(...);
((IAsyncResult)t).AsncWaitHandle.WaitOne();
t.Wait();
```